14_contains/solution/contains-solution.spec.js

```diff
@@ -0,0 +1,60 @@
+const contains = require("./contains-solution");
+
+describe("contains", () => {
+  const meaningOfLifeArray = [42]
+  const object = {
+    data: {
+      duplicate: "e",
+      stuff: {
+        thing: {
+          banana: NaN,
+          moreStuff: {
+            something: "foo",
+            answer: meaningOfLifeArray,
+          },
+        },
+      },
+      info: {
+        duplicate: "e",
+        magicNumber: 44,
+        empty: null,
+      },
+    },
+  };
+
+  test("true if the provided number is a value within the object", () => {
+    expect(contains(object, 44)).toBe(true);
+  });
+
+  test("true if the provided string is a value within the object", () => {
+    expect(contains(object, "foo")).toBe(true);
+  });
+
+  test("does not convert input string into a number when searching for a value within the object", () => {
+    expect(contains(object, "44")).toBe(false);
+  });
+
+  test("false if the provided string is not a value within the object", () => {
+    expect(contains(object, "bar")).toBe(false);
+  });
+
+  test("true if provided string is within the object, even if duplicated", () => {
+    expect(contains(object, "e")).toBe(true);
+  });
+
+  test("false if the provided array has the same structure as an array within the object, however the reference is not the same", () => {
+    expect(contains(object, [42])).toBe(false);
+  });
+
+  test("true if the provided array is a reference to an array that exists within the object", () => {
+    expect(contains(object, meaningOfLifeArray)).toBe(true);
+  });
```

Comment:
Some nits:

- I'd put the same reference test first
- Perhaps a little less wordy on the test descriptions? e.g.
  - "true if the object contains the same object by reference"
  - "false if no matching object reference"

Reply:
sounds good!!